### ISSUE TITLE
Error `ECONNRESET` for validating external links

### ISSUE THREAD

**Issue by amimas** [textlint-rule/textlint-rule-no-dead-link#119]
I have certain links that gets reported as broken due to `ECONNRESET`:

```log
  34:28  error  https://jcp.org/en/jsr/detail?id=330 is dead. (request to https://jcp.org/en/jsr/detail?id=330 failed, reason: connect ECONNRESET 137.254.60.38:443)  no-dead-link
  61:34  error  https://dev.mysql.com/downloads/mysql/ is dead. (request to https://dev.mysql.com/downloads/mysql/ failed, reason: connect ECONNRESET 137.254.60.11:443)  no-dead-link
```

It's consistent so far. They don't get reported as broken link all the time, but most of them they are reported as broken because of the above mentioned `ECONNRESET` error

This is a continuation of #111 where couple of PRs were merged already to address it but unfortunately even with latest version (`4.4.3`), these errors are still persisting.

**Comment by azu** [comment]
`ECONNRESET` is happened when the request is timeout.
The reason come from server side.

- [node.js - Node js ECONNRESET - Stack Overflow](https://stackoverflow.com/questions/17245881/node-js-econnreset)

📝 Maybe It is releated Node.js issue
[ECONNRESET after response · Issue #27916 · nodejs/node](https://github.com/nodejs/node/issues/27916)

I can not reproduce this issue...
